src/zookeeperLock.ts

```diff
@@ -81,49 +81,48 @@ export class ZookeeperLock extends EventEmitter {
         debuglog(JSON.stringify(this.config));
     }
 
+
     /**
      * create a zookeeper client which powers the lock, done when creating a new lock
      * or zk connection expires
      * @returns {Promise<any>}
      */
-    private createClient() : Promise<any> {
+    private createClient() : Promise<any>|any {
         debuglog('creating client');
-        return new Promise<any>((resolve, reject) => {
-            if (this.client != null) {
-                debuglog('client already created');
-                resolve(true);
-            } else {
-                this.config.serverLocator().then((location) => {
-                    let server = location.host;
-                    if (location.port) {
-                        server += ":" + location.port;
-                    }
-                    debuglog(server);
-                    const client = zk.createClient(server, {
-                        sessionTimeout: this.config.sessionTimeout,
-                        spinDelay: this.config.spinDelay,
-                        retries: this.config.retries
-                    });
-
-                    this.client = client;
-
-                    this.client.once('expired', () => {
-                        debuglog('expired');
-                        this.emit(ZookeeperLock.Signals.LOST);
-                        if (this.client) {
-                            this.client.removeAllListeners();
-                            this.removeAllListeners();
-                            this.client = null;
-                        }
-                        this.createClient();
-                    });
+        if (this.client != null) {
+            debuglog('client already created');
+            return Promise.resolve(true);
+        } else {
+            return this.config.serverLocator().then((location) => {
+                let server = location.host;
+                if (location.port) {
+                    server += ":" + location.port;
+                }
+                debuglog(server);
+                const client = zk.createClient(server, {
+                    sessionTimeout: this.config.sessionTimeout,
+                    spinDelay: this.config.spinDelay,
+                    retries: this.config.retries
+                });
 
-                    this.client.once('disconnected', this.reconnect);
+                this.client = client;
 
-                    resolve(true);
+                this.client.once('expired', () => {
+                    debuglog('expired');
+                    this.emit(ZookeeperLock.Signals.LOST);
+                    if (this.client) {
+                        this.client.removeAllListeners();
+                        this.removeAllListeners();
+                        this.client = null;
+                    }
+                    this.createClient();
                 });
-            }
-        });
+
+                this.client.once('disconnected', this.reconnect);
+
+                return true;
```

Comment:
dont you mean `Promise.resolve(true)`? if not then you can make the function return type `Promise<boolean>|boolean`